cmd/cosign/cli/public_key.go

```diff
@@ -54,12 +61,34 @@ EXAMPLES
 			if (*key == "" && *kmsVal == "") || (*key != "" && *kmsVal != "") {
 				return &KeyParseError{}
 			}
-			return GetPublicKey(ctx, *key, *kmsVal, GetPass)
+			// Get private key file.
+			var reader io.Reader
+			if *key != "" {
+				cl := filepath.Clean(*key)
+				if _, err := os.Stat(cl); os.IsNotExist(err) {
+					return fmt.Errorf("missing or invalid key path: %s", cl)
+				} else if err != nil {
+					return err
+				}
+				reader, err := os.Open(cl)
+				if err != nil {
```

Comment:
it seems like this would already cover the missing or invalid path error above:
https://golang.org/pkg/os/